src/hooks/Trades.ts

```diff
@@ -101,10 +121,23 @@ export function useTradeExactOut(currencyIn?: Currency, currencyAmountOut?: Curr
 
   return useMemo(() => {
     if (currencyIn && currencyAmountOut && allowedPairs.length > 0) {
-      return (
+      const trade =
         Trade.bestTradeExactOut(allowedPairs, currencyIn, currencyAmountOut, { maxHops: 3, maxNumResults: 1 })[0] ??
         null
-      )
+
+      const tradeFixed =
+        Trade.bestTradeExactOut(allowedPairs, currencyIn, currencyAmountOut, { maxHops: 1, maxNumResults: 1 })[0] ??
+        null
+
+      // if input amount difference is within threshold, return single hop trade instead - only check if valid trades
+      if (tradeFixed && trade) {
+        const percent = new Percent(trade.inputAmount.raw, tradeFixed.inputAmount.raw)
```

Comment:
this is really hard to wrap my head around, is it correct?

say the 1 hop trade is 100 input, multihop is 99 input
if (99 / 100) > 99.5% return 1 hop
shouldn't we return multihop since it's different by 1%?

would be good to extract this code to a method and add a unit test

Reply:
I just changed this logic a bunch to make it more clear, now comparing the difference to a percent max, or just returning single if the input/output amount is favorable (based on exactIn or exactOut)

Reply:
gunna move to a function and test